File: src/test/java/blackbox/reader/UnreadableReader.java
package blackbox.reader

import java.io.IOException
import java.io.Reader

internal class UnreadableReader : Reader() {
  @Throws(IOException::class)
  override fun read(cbuf: CharArray, off: Int, len: Int): Int {
    throw IOException("Cannot read")
  }

  @Throws(IOException::class)
  override fun close() {
    throw IOException("Cannot close")
  }
}

File: src/test/java/blackbox/reader/UnreadableReader.kt
package blackbox.reader

import java.io.IOException
import java.io.Reader

internal class UnreadableReader : Reader() {

  override fun read(cbuf: CharArray, off: Int, len: Int): Int {
    throw IOException("Cannot read")
  }

  override fun close() {
    throw IOException("Cannot close")
  }
}